src/spec.c

```diff
@@ -1062,12 +1106,17 @@ static int IndexSpec_AddFieldsInternal(IndexSpec *sp, StrongRef spec_ref, ArgsCu
     }
 
     FieldSpec *fs = IndexSpec_CreateField(sp, fieldName, fieldPath);
+    if (!fs) {
+      QueryError_SetErrorFmt(status, QUERY_ELIMIT, "Schema is currently limited to %d fields",
+                             sp->numFields);
+      goto reset;
+    }
```

Comment:
I think we can't get here - we stop at line 1080.
If we can - add coverage. Remove otherwise.

Reply:
Added  coverage